src/interface/AbstractPortal.sol

```diff
@@ -66,40 +63,71 @@ abstract contract AbstractPortal is Initializable, IERC165Upgradeable {
   function bulkAttest(
     AttestationPayload[] memory attestationsPayloads,
     bytes[][] memory validationPayloads
-  ) public payable virtual {
-    // Run all modules for all payloads
+  ) public payable {
     moduleRegistry.bulkRunModules(modules, attestationsPayloads, validationPayloads);
 
     _onBulkAttest(attestationsPayloads, validationPayloads);
 
-    // Register attestations using the attestation registry
     attestationRegistry.bulkAttest(attestationsPayloads, _getAttester());
   }
 
   /**
-   * @notice Revokes the attestation for the given identifier and can replace it by a new one
+   * @notice Replaces the attestation for the given identifier and replaces it with new attestation
+   * @param attestationId the attestation ID to replace
+   * @param attestationPayload the attestation payload to create attestation and register it
+   * @param validationPayloads the payloads to validate via the modules to issue the attestations
+   * @dev Runs all modules for the portal and registers the attestation using AttestationRegistry
+   */
+  function replace(
+    bytes32 attestationId,
+    AttestationPayload memory attestationPayload,
+    bytes[] memory validationPayloads
+  ) public payable {
+    moduleRegistry.runModules(modules, attestationPayload, validationPayloads, msg.value);
+
+    _onReplace(attestationId, attestationPayload);
+
+    attestationRegistry.replace(attestationId, attestationPayload, _getAttester());
+  }
+
+  /**
+   * @notice Bulk replaces the attestation for the given identifiers and replaces them with new attestations
+   * @param attestationIds the attestation IDs to replace
+   * @param attestationsPayloads the payloads to attest
+   * @param validationPayloads the payloads to validate via the modules to issue the attestations
+   */
+  function bulkReplace(
+    bytes32[] memory attestationIds,
+    AttestationPayload[] memory attestationsPayloads,
+    bytes[][] memory validationPayloads
+  ) public payable {
+    moduleRegistry.bulkRunModules(modules, attestationsPayloads, validationPayloads);
```

Comment:
I think it's better to have a validation on number of attestationIds = number of attestation payloads